src/body/pinned.rs

```diff
@@ -21,7 +21,7 @@ impl Pinned {
     /// This method is marked as unsafe because the caller must ensure that the
     /// provided body is `!Unpin` and properly remains pinned in memory while the
     /// body is polled.
-    pub unsafe fn new_unchecked<B>(body: Box<B>) -> Self
```

Comment:
I decided not to make this `unsafe` since the caller presumably wrote a pin projection using `unsafe` in order to create the `!Unpin` body. I'm curious to hear how the community feels about this.